opal/mca/threads/qthreads/configure.m4

```diff
@@ -25,13 +25,67 @@
 
 AC_DEFUN([OPAL_CONFIG_QTHREADS],[
 
-    AC_CHECK_HEADERS([qthread/qthread.h],
-                     [AC_CHECK_LIB([qthread],[qthread_initialize],
-                                    [threads_qthreads_happy="yes"],
-                                    [threads_qthreads_happy="no"])],
-                     [threads_qthreads_happy="no"])
+    AC_ARG_WITH([qthreads],
+                [AC_HELP_STRING([--with-qthreads=DIR],
+                                [Specify location of qthreads installation.  Error if qthreads support cannot be found.])])
 
-    AS_IF([test "$threads_qthreads_happy" = "yes"],
+    AC_ARG_WITH([qthreads-libdir],
+                [AC_HELP_STRING([--with-qthreads-libdir=DIR],
+                                [Search for qthreads libraries in DIR])])
+
+    opal_check_qthreads_save_CPPFLAGS=$CPPFLAGS
+    opal_check_qthreads_save_LDFLAGS=$LDFLAGS
+    opal_check_qthreads_save_LIBS=$LIBS
+
+    opal_qthreads_happy=yes
+    AS_IF([test "$with_qthreads" = "no"],
+          [opal_qthreads_happy=no])
+
+    AS_IF([test $opal_qthreads_happy = yes],
+          [AC_MSG_CHECKING([looking for qthreads in])
+           AS_IF([test "$with_qthreads" != "yes"],
+                 [opal_qthreads_dir=$with_qthreads
+                  AC_MSG_RESULT([($opal_qthreads_dir)])],
+                 [AC_MSG_RESULT([(default search paths)])])
+           AS_IF([test ! -z "$with_qthreads_libdir" && \
+                         test "$with_qthreads_libdir" != "yes"],
+                 [opal_qthreads_libdir=$with_qthreads_libdir])
+          ])
+
+    AS_IF([test $opal_qthreads_happy = yes],
+          [OPAL_CHECK_PACKAGE([opal_qthreads],
+                              [qthread.h],
+                              [qthread],
+                              [qthread_initialize],
+                              [],
+                              [$opal_qthreads_dir],
+                              [$opal_qthreads_libdir],
+                              [],
+                              [opal_qthreads_happy=no])])
+
+    AS_IF([test $opal_qthreads_happy = yes && test -n "$opal_qthreads_dir"],
+          [OPAL_QTHREADS_INCLUDE_PATH="$opal_qthreads_dir/include/"],
+          [OPAL_QTHREADS_INCLUDE_PATH=""])
+
+    AS_IF([test $opal_qthreads_happy = yes],
+          [AC_CONFIG_FILES([opal/mca/threads/qthreads/threads_qthreads.h])
+           AC_SUBST([OPAL_QTHREADS_INCLUDE_PATH])
+           AC_SUBST([opal_qthreads_CPPFLAGS])
+           AC_SUBST([opal_qthreads_LDFLAGS])
+           AC_SUBST([opal_qthreads_LIBS])
```

Comment:
I believe that AC_CONFIG_FILES and AC_SUBST are always executed, regardless of the result of the AS_IF.  I.e., these should probably be outside the AS_IF so as not to imply that they are actually conditionally executed.

Reply:
I moved them outside `AS_IF()`.

> AC_CONFIG_FILES and AC_SUBST are always executed

Yes they seem so. I did not know that. Thank you!